chipsec/modules/tools/uefi/reputation.py

```diff
@@ -0,0 +1,142 @@
+"""
+This module checks current contents of UEFI firmware ROM or specified firmware image for bad EFI binaries as per the
+VirusTotal API. These can be EFI firmware volumes, EFI executable binaries (PEI modules, DXE drivers..) or EFI sections.
+The module can find EFI binaries by their UI names, EFI GUIDs, MD5/SHA-1/SHA-256 hashes
+or contents matching specified regular expressions.
+
+Important! This module can only detect bad or vulnerable EFI modules based on the file's reputation on VT.
+
+Usage:
+    chipsec_main.py -i -m tools.uefi.reputation -a <vt_api_key>[,<vt_threshold>,<fw_image>]
+      vt_api_key   : API key to VirusTotal. Can be obtained by visting https://www.virustotal.com/gui/join-us.
+                     This argument must be specified.
+      vt_threshold : The minimal number of different AV vendors on VT which must claim an EFI module is malicious
+                     before failing the test. Defaults to 10.
+      fw_image     : Full file path to UEFI firmware image
+                     If not specified, the module will dump firmware image directly from ROM
+"""
+import time
+
+from chipsec.module_common import BaseModule, ModuleResult, MTAG_BIOS
+from chipsec.hal.spi_uefi import search_efi_tree, build_efi_model, EFIModuleType
+from chipsec.hal.uefi import UEFI
+from chipsec.hal.spi import SPI, BIOS
+from chipsec.file import read_file
+
+try:
+    from virus_total_apis import PublicApi as VirusTotalPublicApi
+    has_virus_total_apis = True
+except ImportError:
+    has_virus_total_apis = False
+
+TAGS = [MTAG_BIOS]
+
+DEF_FWIMAGE_FILE = 'fw.bin'
+
+USAGE_TEXT = '''
+Usage:
+
+    chipsec_main.py -i -m tools.uefi.reputation -a <vt_api_key>[,<vt_threshold>,<fw_image>]
+
+      vt_api_key   : API key to VirusTotal. Can be obtained by visiting https://www.virustotal.com/gui/join-us.
+                     This argument must be specified.
+      vt_threshold : The minimal number of different AV vendors on VT which must claim an EFI module is bad
+                     before failing the test. Defaults to 10.
+      fw_image     : Full file path to UEFI firmware image
+                     If not specified, the module will dump firmware image directly from ROM
+'''
+
+class reputation(BaseModule):
+
+    def __init__(self):
+        BaseModule.__init__(self)
+        self.uefi = UEFI( self.cs )
+        self.image = None
+        self.vt_threshold = 10
+        self.vt = None
+
+    def is_supported(self):
+        if has_virus_total_apis:
+            return True
+        else:
+            self.logger.log_important("""Can't import module 'virus_total_apis'.
+Please run 'pip install virus_total_apis' and try again.""")
```

Comment:
Per [pypi](https://pypi.org/project/virustotal-api/), it should be:
```bash
pip install virustotal-api
```

Reply:
My bad, fixed.